src/mame/drivers/mpu4unsorted.cpp

```diff
@@ -0,0 +1,2598 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+/*
+    These are MPU4 based machines that are still in need of correct placement.
+
+    Some of them appear to be official Barcrest machines, and will likely end up
+    being moved to the mod2/mod4 files based on hardware types.
+
+    Official machines / licensed machines can typically be identified by the
+    standard Barcrest 'characteriser' protection sequence being written.
+
+    Other machines in this files are unofficial machines, maybe one-off units
+    that will require individual attention.
+*/
+
+#include "emu.h"
+#include "includes/mpu4.h"
+
+
+// in mpu4.cpp
+INPUT_PORTS_EXTERN( mpu4 );
+INPUT_PORTS_EXTERN( mpu4_invcoin );
+INPUT_PORTS_EXTERN( mpu420p );
+INPUT_PORTS_EXTERN( mpu4_cw );
+INPUT_PORTS_EXTERN( mpu4jackpot8tkn );
+INPUT_PORTS_EXTERN( mpu4jackpot8per );
+INPUT_PORTS_EXTERN( grtecp );
```

Comment:
Same here – all these should just be declared in includes/mpu4.h if they’re defined in mpu4.cpp